src/libraries/Microsoft.Extensions.Diagnostics/src/Metrics/DefaultMeterFactory.cs

```diff
@@ -62,6 +107,168 @@ public Meter Create(MeterOptions options)
             }
         }
 
+        [MemberNotNull(nameof(_rules))]
+        private void UpdateRules(MetricsEnableOptions options, string? name)
+        {
+            lock (_connections)
+            {
+                _rules = options.Rules;
+
+                if (_disposed)
+                {
+                    return;
+                }
+
+                foreach (var pair in _connections)
+                {
+                    RefreshConnections(pair.Key, pair.Value);
+                }
+            }
+        }
+
+        private void InstrumentPublished(Instrument instrument, MeterListener listener)
+        {
+            lock (_connections)
+            {
+                if (_disposed)
+                {
+                    return;
+                }
+
+                if (_connections.ContainsKey(instrument))
+                {
+                    Debug.Assert(false, "InstrumentPublished called twice for the same instrument");
+                    return;
+                }
+
+                var listeners = _connections.GetOrAdd(instrument, static _ => new());
+                RefreshConnections(instrument, listeners);
+            }
+        }
+
+        // Called under _connections lock
+        private void RefreshConnections(Instrument instrument, ConcurrentDictionary<IMetricsListener, object?> listeners)
+        {
+            // Find any that match, pair them
+            var newListeners = new HashSet<IMetricsListener>();
+            var alreadyListening = !listeners.IsEmpty;
+            foreach (var rule in _rules)
+            {
+                if (RuleMatchesInstrument(rule, instrument))
```

Comment:
I forgot to comment before, logging has a set of precedence heuristics when multiple rules cover the same logger. Check out `LoggerRuleSelector`. I was assuming we'd replicate that logic probably with one extra selection priority for scope. I'd suggest local scope take precedence over global scope, but as long as it is well defined I don't think it matters. I'm not expecting users to create rules that rely on the subtleties under any usual circumstances.